api/client/proxy/proxy_test.go

```diff
@@ -109,14 +110,42 @@ func TestGetProxyAddress(t *testing.T) {
 	for i, tt := range tests {
 		t.Run(fmt.Sprintf("%v: %v", i, tt.info), func(t *testing.T) {
 			for _, env := range tt.env {
-				t.Setenv(env.name, env.val)
+				if strings.EqualFold(env.name, "http_proxy") || strings.EqualFold(env.name, "https_proxy") {
+					t.Setenv(env.name, fmt.Sprintf(env.val, "")) // test without user:pass@addr form
+				} else {
+					t.Setenv(env.name, env.val)
+				}
 			}
-			p := GetProxyAddress(tt.targetAddr)
+			p := GetProxyURL(tt.targetAddr)
 			if tt.proxyAddr == "" {
 				require.Nil(t, p)
 			} else {
 				require.NotNil(t, p)
 				require.Equal(t, tt.proxyAddr, p.Host)
+				require.Nil(t, p.User)
+			}
+
+			// now reconstruct the env var with user:pass@host, set the env again, and test parsing that
+			expectedUser := "alice"
+			expectedPassword := "password"
+			creds := fmt.Sprintf("%v:%v@", expectedUser, expectedPassword)
+			for _, env := range tt.env {
+				if strings.EqualFold(env.name, "http_proxy") || strings.EqualFold(env.name, "https_proxy") {
```

Comment:
At first glance this look like a separate test case could you extract this logic to a separate test function ? . Also what would you say about reusing `url.Parse` -> `u.String` to set the user and password instead of using "%v" template ?

Reply:
I wanted to re-use these test cases, that's why I didn't make a separate test function. We can use url.Parse but only when the proxy addr is not in the form `host:port`, as it won't parse correctly as a valid url. I'll update this